chain/solana/solana.go

```diff
@@ -2,22 +2,47 @@ package solana
 
 import (
 	"context"
+	"encoding/binary"
 	"encoding/json"
 	"fmt"
 
 	"github.com/btcsuite/btcutil/base58"
+	"github.com/renproject/multichain"
 	"github.com/renproject/multichain/api/address"
+	"github.com/renproject/multichain/api/contract"
 	"github.com/renproject/pack"
+	"github.com/renproject/surge"
 	"go.uber.org/zap"
 )
 
+var (
+	// GatewayRenBTC is the address of the Gateway program deployed on Solana
+	// for the selector "BTC/toSolana".
+	GatewayRenBTC = "9TaQuUfNMC5rFvdtzhHPk84WaFH3SFnweZn4tw9RriDP"
+)
+
+// GatewayProgram returns the address of the deployed Gateway program for the
+// given asset.
+func GatewayProgram(asset multichain.Asset) (multichain.Address, error) {
```

Comment:
@roynalnaruto why do we have this function? Seems inflexible and highly application specific. It also doesn't really do anything.